src/main/java/build/buildfarm/cas/cfc/CASFileCache.java

```diff
@@ -2585,6 +2602,26 @@ private CancellableOutputStream putOrReference(
     }
   }
 
+  private void deleteExpiredKey(String expiredKey) throws IOException {
+    Path path = getPath(expiredKey);
+
+    // We don't want the metric publishing to delay the deletion of the file.
+    // We publish the metric only after the file has been deleted.
+    if (publishTtlMetric) {
+      long createdTime = path.toFile().lastModified();
+      Files.delete(path);
+      publishExpirationMetric(createdTime);
+    } else {
+      Files.delete(path);
+    }
+  }
```

Comment:
Would prefer two blocks checking publishTtlMetric for the population and then publish to two copies of the delete call here.

Reply:
fixed

Reply:
I think is correct now.  I still have to get the modifed time, so I keep that outside the two blocks.